numpy/_core/shape_base.py

```diff
@@ -455,6 +457,77 @@ def stack(arrays, axis=0, out=None, *, dtype=None, casting="same_kind"):
     return _nx.concatenate(expanded_arrays, axis=axis, out=out,
                            dtype=dtype, casting=casting)
 
+def _unstack_dispatcher(x, *, axis=None):
+    return (x,)
+
+@array_function_dispatch(_unstack_dispatcher)
+def unstack(x, /, *, axis=0):
+    """
+    Splits an array into a sequence of arrays along the given axis.
+
+    The ``axis`` parameter specifies the axis along which the array will be
+    split. of the new axis in the dimensions of the result. For example, if
+    ``axis=0`` it will be the first dimension and if ``axis=-1`` it will be
+    the last dimension.
+
+    The result is a tuple of arrays split along ``axis``. ``unstack`` serves
+    as the reverse operation of :py:func:`stack`, i.e., ``stack(unstack(x,
+    axis=axis), axis=axis) == x``.
+
+    .. versionadded:: 2.1.0
+
+    Parameters
+    ----------
+    x : ndarray
+        The array to be unstacked.
+
+    Returns
+    -------
+    unstacked : tuple of ndarrays
+        The unstacked arrays.
+
+    See Also
+    --------
+    stack : Join a sequence of arrays along a new axis.
+    concatenate : Join a sequence of arrays along an existing axis.
+    vstack : Stack arrays in sequence vertically (row wise).
+    hstack : Stack arrays in sequence horizontally (column wise).
+    dstack : Stack arrays in sequence depth wise (along third axis).
+    column_stack : Stack 1-D arrays as columns into a 2-D array.
+    vsplit : Split an array into multiple sub-arrays vertically (row-wise).
+    unstack : Split an array into a tuple of sub-arrays along an axis.
```

Comment:
Should not do `see also` to oneself! In general, I'd tend to remove quite a number of the items here: `[vhd]stack`, definitely `column_stack`, and probably `vsplit`.